specification/advisor/resource-manager/Microsoft.Advisor/preview/2025-08-01-preview/advisor.json

```diff
@@ -0,0 +1,1992 @@
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted. Recommendation generation has been accepted.",
+            "headers": {
+              "Location": {
+                "description": "The URL where the status of the asynchronous operation can be checked.",
+                "type": "string"
+              },
+              "Retry-After": {
+                "description": "The amount of delay to use while the status of the operation is checked. The value is expressed in seconds.",
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ArmErrorResponse"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-examples": {
+          "GenerateRecommendations": {
+            "$ref": "./examples/GenerateRecommendations.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Advisor/generateRecommendations/{operationId}": {
+      "get": {
+        "tags": [
+          "GenerateRecommendations"
+        ],
+        "description": "Retrieves the status of the recommendation computation or generation process. Invoke this API after calling the generation recommendation. The URI of this API is returned in the Location field of the response header.",
+        "operationId": "Recommendations_GetGenerateStatus",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "name": "operationId",
+            "in": "path",
+            "description": "The operation ID, which can be found from the Location field in the generate recommendation response header.",
+            "required": true,
+            "type": "string",
+            "format": "uuid"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted. Recommendation generation is in progress."
+          },
+          "204": {
+            "description": "NoContent. Recommendation generation has been completed."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ArmErrorResponse"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-examples": {
+          "GetGenerateStatus": {
+            "$ref": "./examples/EmptyResponse.json"
+          }
+        }
+      }
+    },
+    "/{resourceUri}/providers/Microsoft.Advisor/recommendations": {
```

Comment:
>resourceUri [](http://example.com/codeflow?start=7&length=11)

So, this can work on any azure resource scope ? want to make sure if the RP is ready for this.